plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/LogOutputStream.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2022 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.model.preferences.DBPPreferenceListener;
+import org.jkiss.dbeaver.model.preferences.DBPPreferenceStore;
+import org.jkiss.dbeaver.runtime.DBWorkbench;
+
+public class LogOutputStream extends OutputStream {
+
+    public static final long DEFAULT_MAX_LOG_SIZE = 1024 * 1024 * 10; // 10Mb    
+    public static final int DEFAULT_MAX_LOG_FILES_COUNT = 3;
+    
+    public static final String LOGS_MAX_FILE_SIZE = "logs.files.output.maxSize";
+    public static final String LOGS_MAX_FILES_COUNT = "logs.files.output.maxCount";
+
+
+    /**
+     * The File object to store messages.  This value may be null.
+     */
+    private final File currentLogFile;
+    
+    private final File logFileLocation;
+    
+    private final DBPPreferenceStore prefStore;
+
+    /**
+     * The Writer to log messages to.
+     */
+    private volatile FileOutputStream currentLogFileOutput = null;
+    private volatile long currentLogSize;
+
+    private volatile long maxLogSize;
+    private volatile int maxLogFiles;
+    
+    private final String logFileName;
+    private final String logFileNameExtension;
+    private final Predicate<String> logFileNamePattern;
+    
+    public LogOutputStream(@NotNull File debugLogFile) throws IOException {
+        if (debugLogFile.exists() && !debugLogFile.isFile()) {
+            throw new IOException(
+                "Failed to initialize debug log output due to the target is not a file: " + debugLogFile.getAbsolutePath()
+            );
+        }
+        
+        this.currentLogFile = debugLogFile;
+        this.logFileLocation = debugLogFile.getParentFile();
+        this.prefStore = DBWorkbench.getPlatform().getPreferenceStore();
```

Comment:
This field is not used outside the constructor. Can it be a local variable instead?

Or maybe you wanted to remove the property change listener from it in the `close` method?